File: aesi-eclipse/bundle/src/main/java/com/virtlink/aesi/structureoutline/IStructureOutliner.java
package com.virtlink.aesi.structureoutline;

import com.virtlink.aesi.IAesiDocument;
import com.virtlink.aesi.ICancellationToken;

import javax.annotation.Nullable;
import java.util.List;

/**
 * Provides an outline of the document structure.
 */
public interface IStructureOutliner {

    List<? extends ISymbol> getRoots(IAesiDocument document,
                                     @Nullable ICancellationToken cancellationToken);

    List<? extends ISymbol> getChildren(IAesiDocument document,
                                        ISymbol symbol,
                                        @Nullable ICancellationToken cancellationToken);

    boolean hasChildren(IAesiDocument document,
                        ISymbol symbol,
                        @Nullable ICancellationToken cancellationToken);

}

File: aesi/src/main/kotlin/com/virtlink/editorservices/structureoutline/IStructureOutliner.kt
package com.virtlink.editorservices.structureoutline

import com.virtlink.editorservices.ICancellationToken
import com.virtlink.editorservices.IDocument
import com.virtlink.editorservices.IDocumentLocation
import com.virtlink.editorservices.IProject
import com.virtlink.editorservices.codecompletion.ICompletionProposal

@Deprecated("Replaced by IStructureOutlineService")
interface IStructureOutliner {

    /**
     * Returns the child symbols of the given symbol (if any)
     * or the root of the document.
     */
    fun outline(project: IProject,
                document: IDocument,
                symbol: ISymbol?,
                cancellationToken: ICancellationToken?)
            : List<ISymbol>

}